x-pack/plugin/spatial/src/main/java/org/elasticsearch/xpack/spatial/search/aggregations/bucket/geogrid/AbstractGeoHexGridTiler.java

```diff
@@ -19,6 +19,15 @@
 abstract class AbstractGeoHexGridTiler extends GeoGridTiler {
 
     private static final long[] RES0CELLS = H3.getLongRes0Cells();
+    // The hexRing neighbours optimization is insufficient at the Poles, so we do a little extra checking
+    private static final long[] NORTH_POLAR_CELLS = new long[H3.MAX_H3_RES];
```

Comment:
I think this information is already in H3CartesianUtils, cannot we add just methods on that class?

Reply:
Actually there is a method -> H3CartesianUtils.#isPolar that does what you want.